builtin/providers/aws/data_source_aws_ami.go

```diff
@@ -193,7 +193,11 @@ func dataSourceAwsAmiRead(d *schema.ResourceData, meta interface{}) error {
 		params.Filters = buildAwsDataSourceFilters(filters.(*schema.Set))
 	}
 	if ownersOk {
-		params.Owners = expandStringList(owners.([]interface{}))
+		o := expandStringList(owners.([]interface{}))
+
+		if len(o) > 0 {
+			params.Owners = o
```

Comment:
Added a guard here for when Owners is empty but sent, which throws:

    MissingParameter: The request must contain the parameter Owner